scripts/hudson/narayana.sh

```diff
@@ -123,11 +123,41 @@ function check_if_pull_closed
 function build_narayana {
   echo "Building Narayana"
   cd $WORKSPACE
-  [ $NARAYANA_TESTS = 1 ] && NARAYANA_ARGS= || NARAYANA_ARGS="-DskipTests"
 
-  ./build.sh -Prelease,community,all$OBJECT_STORE_PROFILE -Didlj-enabled=true "$@" $NARAYANA_ARGS $IPV6_OPTS -B clean install
+  # JAVA_HOME_FOR_BUILD can be used in environment if JDK for building Narayana needs to be different from one which runs e.g. tests.
+  # This is an option used for testing on JDK6. The Narayana 4.17 requires JDK7 for building but target version is JDK6.
```

Comment:
The pom says 1.6 is used for the source and target which doesn't match this statement:
```
         <artifactId>maven-compiler-plugin</artifactId>
          <version>2.3.2</version>
          <configuration>
            <source>1.6</source>
            <target>1.6</target>
            <compilerArgument>-AtranslationFilesPath=${project.basedir}/classes/</compilerArgument>
          </configuration>
```

Do we need to change the <source>1.6</source> parameter?

Reply:
That's a good point. The code can't be build on JDK6 as the code uses methods which are not available in JDK6 (e.g. suppressed exceptions). On the other side the code does not use any JDK7 specific language syntax. That way the source is 1.6 compatible but cannot be built on JDK6. It's a question how we want to approach it. Maybe changing to 1.7 would be better for guiding the user. I'm going to change it.

Reply:
Ok, my wrong. The `maven-compiler-plugin` requires the `source` being the same (or lower probably) than the `target`. As we need to build for JDK6 then the `source` parameter *must* be 1.6.

Reply:
My point is that the new comment contradicts what the pom says. The pom says source is 1.6 whereas your new comment says that "Narayana 4.17 requires JDK7 for building".

Reply:
So the pom says we can only use JDK 1.6 features in the source.
Do we require JDK 1.7 for compiling because our version of maven requires it or that we have some pom dependencies that are incompatible with JDK1.6?

Reply:
For clarification, the new comment in the code means:
```
We target 1.6 byte code.
And the java source is limited to 1.6 language features only.

But we must build on 1.7 because we are using a JDK API that only exists in JDK 1.7.

That is what we mean by "Narayana 4.17 requires JDK7 for building"?
```

Reply:
+1, I'm going to adjust the comment to be more understandable.